### ISSUE TITLE
Declared dependencies for components require run-time layerUI.init()

### ISSUE THREAD

**Issue by EmielM** [layerhq/layer-ui-web#4]
Hey,

I'm using `rollup` to build my app in IIFE form. It builds a dependency graph, and then simply evaluates the files in order.

This doesn't work as I would want to because `layer-ui-web` components do as follows:

`import { layer as LayerAPI } from '../../base';`

In short: they depend on the run-time `layerUI.init({layer: xx})` call. That might be (barely) forgivable when `import` was still `require()`, but right now it simply breaks the dependency declarations that es6 imports are here to solve.

My current work-around is the following:

```typescript
// layer0.ts
import layer from 'layer-websdk';
import layerUI from 'layer-ui-web/lib/base';

layerUI.init({ layer: layer, appId: layerAppId });

export {layerUI, layer, layerAppId};
```

```typescript
// layer.ts
import {layer, layerUI, layerAppId} from './layer0'

import 'layer-ui-web/lib/components/layer-conversation-panel/layer-conversation-panel';
import 'layer-ui-web/lib/components/messages-list-panel/layer-message-item/layer-message-item';
import 'layer-ui-web/lib/components/messages-list-panel/layer-messages-list/layer-messages-list';
// a gazillion other files that I need to add here, why can't inter-component dependencies be declared in the files?
// which ones? trial and error! the list in Readme is out of date
```

But that's all very hacky, not very maintainable and might still break because [`layer0` first, `component`s later] is still not properly declared (eg: rollup might decide to not execute layer0 before the component).

Why doesn't `layer-ui-web` just `import LayerAPI from 'layer-websdk';`?

And why aren't dependencies on sub-components declared? To allow customization? I would rather do this by patching layer-ui-web, or tweaking my bundler to use alternative files.

**Comment by michaelkantor** [comment]
Ok, a bunch of questions here, and I don't fully grasp the nuances of all of them.  Let me start with the easy ones:

1. `layer-ui-web` does not import from `layer-websdk` because it has no way of knowing if you've loaded the WebSDK via `npm`, `cdn`, if its part of a build file, or something else.  To work around this, `layer` is a key input to `layerUI.init()`.
2. Dependencies on sub-components are not declared so that you can easily replace our UI components with your own.  We anticipate working on a pre-bundled, non-customized library that assembles all these widgets into a UI for you.  But this library is the  core UI widgets, and must remain customizable.  If you've seen our latest update, (https://docs.layer.com/sdk/webui/ui_customization#modify-components) customization of widgets is a core goal.
3. I have no experience with `rollup` and its hard for me to comment on best practices or what is forgivable or not based on a tool and strategy I've never used.  What I can say is the UI components are unable to import `layer` in the component's `import` section, which is incredibly annoying, and could be addressed if I could find a better way to do the whole `layerUI.init()` thing.  So, while I do NOT understand issues with `rollup`, I'm more than happy to discuss alternative approaches to my comment in item 1 above.

michael

**Comment by michaelkantor** [comment]
https://github.com/layerhq/layer-ui-web/commit/75cf0324e6f272cfe91c516672f4722bf521f989

Shipped as Layer UI for Web 0.9.10.
